include/deal.II/lac/read_write_vector.h

```diff
@@ -517,6 +535,35 @@ namespace LinearAlgebra
      * Make all other ReadWriteVector types friends.
      */
     template <typename Number2> friend class ReadWriteVector;
+
+#ifdef DEAL_II_WITH_CXX11
+  private:
+    template <typename Functor>
+    class FunctorTemplate
```

Comment:
can you add a brief documentation to this class, indicating what it may be used for?